druid/src/core.rs

```diff
@@ -1177,6 +1190,10 @@ impl WidgetState {
     pub(crate) fn add_sub_window_host(&mut self, window_id: WindowId, host_id: WidgetId) {
         self.sub_window_hosts.push((window_id, host_id))
     }
+
+    pub(crate) fn window_origin(&self) -> Point {
+        self.parent_window_origin + self.origin.to_vec2() - self.viewport_offset
```

Comment:
is this only used from the various contexts? Could we get away with just storing `parent_window_origin` once, in the `ContextState`, and doing this addition there?

Reply:
I think it’s much easier to communicate this type of thing in a follow up PR

Reply:
I don't understand how this would work tbh (wouldn't it end up doing calcs that it doesn't need in lots of contexts?) so I'll leave it for now. I did previously store the window origin but that became annoying for viewport_offset updates and had to fiddle with the internal event on the way down.... I'm sure there are ways to rationalise this later when its in place. 

I think viewport_offset in the end wants to be a general invertible transformation (possibly more general than Affine, eg if you wanted perspective or 3d projection etc). Anyway off topic for this...